### PR TITLE
Make sure all project references are distinct

### PR DESCRIPTION
**Pull request by Mingxue008**
This is to resolve issue #203.

### CODE REVIEW

**Comment by birojnayak** [comment]
looks good to me

File: src/Buildalyzer/AnalyzerResult.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ internal AnalyzerResult(string projectFilePath, AnalyzerManager manager, Project
         public IEnumerable<string> ProjectReferences =>
             Items.TryGetValue("ProjectReference", out IProjectItem[] items)
                 ? items.Select(x => AnalyzerManager.NormalizePath(
-                    Path.Combine(Path.GetDirectoryName(ProjectFilePath), x.ItemSpec)))
+                    Path.Combine(Path.GetDirectoryName(ProjectFilePath), x.ItemSpec))).Distinct()
```
Comment (Therzok) [inline src/Buildalyzer/AnalyzerResult.cs:103]: Should case sensitivity be taken into account here?

Comment (Mingxue008) [reply src/Buildalyzer/AnalyzerResult.cs]: Thanks for catching that, should be resolved with second commit.
Also added test case for this change.

**Comment by daveaglick** [comment]
Thanks @Mingxue008 for the fix! And thanks @birojnayak and @Therzok for the review! I'll get this merged and released tomorrow 👍

File: src/Buildalyzer.Workspaces/AnalyzerResultExtensions.cs [hunk 2]
```diff
@@ -254,7 +254,7 @@ private static CompilationOptions CreateCompilationOptions(IAnalyzerResult analy
 
         private static IEnumerable<ProjectReference> GetExistingProjectReferences(IAnalyzerResult analyzerResult, Workspace workspace) =>
             analyzerResult.ProjectReferences
-                .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath == x))
+                .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.CurrentCultureIgnoreCase)))
```
Comment (slang25) [inline src/Buildalyzer.Workspaces/AnalyzerResultExtensions.cs:257]:
```suggestion
                .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.OrdinalIgnoreCase)))

```

File: src/Buildalyzer/AnalyzerResult.cs [hunk 3]
```diff
@@ -370,8 +371,8 @@ private static IEnumerable<string> EnumerateCommandLinePartsFsc(string commandLi
 
         private class ProjectItemItemSpecEqualityComparer : IEqualityComparer<IProjectItem>
         {
-            public bool Equals(IProjectItem x, IProjectItem y) => x.ItemSpec.Equals(y.ItemSpec);
-            public int GetHashCode(IProjectItem obj) => obj.ItemSpec.GetHashCode();
+            public bool Equals(IProjectItem x, IProjectItem y) => x.ItemSpec.Equals(y.ItemSpec, StringComparison.CurrentCultureIgnoreCase);
```
Comment (slang25) [inline src/Buildalyzer/AnalyzerResult.cs:374]:
```suggestion
            public bool Equals(IProjectItem x, IProjectItem y) => x.ItemSpec.Equals(y.ItemSpec, StringComparison.OrdinalIgnoreCase);

```

File: src/Buildalyzer/AnalyzerResult.cs [hunk 4]
```diff
@@ -370,8 +371,8 @@ private static IEnumerable<string> EnumerateCommandLinePartsFsc(string commandLi
 
         private class ProjectItemItemSpecEqualityComparer : IEqualityComparer<IProjectItem>
         {
-            public bool Equals(IProjectItem x, IProjectItem y) => x.ItemSpec.Equals(y.ItemSpec);
-            public int GetHashCode(IProjectItem obj) => obj.ItemSpec.GetHashCode();
+            public bool Equals(IProjectItem x, IProjectItem y) => x.ItemSpec.Equals(y.ItemSpec, StringComparison.CurrentCultureIgnoreCase);
+            public int GetHashCode(IProjectItem obj) => obj.ItemSpec.ToLower().GetHashCode();
```
Comment (slang25) [inline src/Buildalyzer/AnalyzerResult.cs:375]:
```suggestion
            public int GetHashCode(IProjectItem obj) => obj.ItemSpec.ToLowerInvariant().GetHashCode();

```

Comment (daveaglick) [reply src/Buildalyzer/AnalyzerResult.cs:375]: Yep, I agree with @slang25 we should be using `OrdinalIgnoreCase` everywhere `CurrentCultureIgnoreCase` was added. Nice catch. Also agree with `ToLowerInvariant`.

**Comment by daveaglick** [comment]
FYI - I went ahead and applied the suggestions to the PR branch to get a CI build going. Assuming that passes I'll merge it all in.

**Comment by daveaglick** [comment]
Thanks all! This will go out in the next version (hopefully today, but also trying to track down another issue I'd like to fix in the release so might get delayed).

**Pull request merged**